src/Leaflet.Renderer.SVG.Tile.js

```diff
@@ -39,6 +39,16 @@ L.SVG.Tile = L.SVG.extend({
 		}
 	},
 
+	removeFrom: function (map) {
+		if (this.options.interactive) {
+			for (var i in this._layers) {
+				var layer = this._layers[i];
+				delete this._map._targets[L.stamp(layer._path)];
```

Comment:
Yep, Leaflet's vector rendering is quite intricate as it is, and it's no easier with the trickery involved in VectorGrid. We should think about if there's a way to make this easier, preferably without re-implementing Leaflet's vector functionality.